code/game/objects/items/dyekit.dm

```diff
@@ -0,0 +1,33 @@
+/obj/item/dyespray
+	name = "hair dye spray"
+	desc = "A spray to dye your hair any gradients you'd like."
+	icon = 'icons/obj/dyespray.dmi'
+	icon_state = "dyespray"
+
+/obj/item/dyespray/attack_self(mob/user)
+	dye(user)
+
+/obj/item/dyespray/pre_attack(atom/target, mob/living/user, params)
+	dye(target)
+	return ..()
+
+/obj/item/dyespray/proc/dye(mob/target)
+	if(!ishuman(target))
+		return
+	var/mob/living/carbon/human/human_target = target
+
+	var/new_grad_style = input(usr, "Choose a color pattern:", "Character Preference")  as null|anything in GLOB.hair_gradients_list
+	if(!new_grad_style)
+		return
+
+	var/new_grad_color = input(usr, "Choose a secondary hair color:", "Character Preference","#"+human_target.grad_color) as color|null
+	if(!new_grad_color)
+		return
+
+	human_target.grad_style = new_grad_style
+	human_target.grad_color = sanitize_hexcolor(new_grad_color)
+	to_chat(human_target, "<span class='notice'>You apply the hair dye...</span>")
```

Comment:
grammar nitpick, should reword this to imply you're starting to take action, not that you took the action, as is the norm for do_after actions

Reply:
Fixed!